### PR TITLE
feat: allow cookie option overrides on save

### PR DESCRIPTION
**Pull request by jrobber**
Save now takes an optional parameter that is another set of cookie options.

If there are any there they override the default options passed in during initial creation.

Use Case:
* I apply a basic session to a user
* User logs into facebook or other provider
* Facebook dictates a max age/expires length
-> when I update my session I can also update maxAge to match what facebook provided

Also, this generic change should support a whole host of other scenarios and cookie management.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "build": "babel lib/ -d dist/ --source-maps --ignore '**/*.test.js' --delete-dir-on-start",
     "format": "prettier --write '**/*.*' && eslint . --fix",
     "lint": "prettier --check '**/*.*' && eslint .",
+    "prepare": "yarn build",
```
Comment (vvo) [inline package.json:24]: Hi there, any comment you can provide about this change?

File: lib/index.js [hunk 2]
```diff
@@ -59,9 +59,12 @@ export async function applySession(
     set: store.set,
     get: store.get,
     unset: store.unset,
-    async save() {
+    async save(cookieOptionsOverrides = {}) {
```
Comment (vvo) [inline lib/index.js:62]: Let's use the same option name as init: `.save({cookieOptions: {...}})`, can you make this change?

**Review by vvo — CHANGES_REQUESTED** [review]
Hi there, thanks for the suggestion, it makes sense. So that I can accept this PR we would need:
- API documentation update
- typings update
- add a simple test

Let me know which task you think is OK and which one you'd like help with, thanks

**Comment by pedrojimenezp** [comment]
@vvo @jrobber  Maybe I can continue the work of this PR if you don't have time.
I can work on the fixes/task you are asking for so you can merge it soon.

**Comment by vvo** [comment]
@pedrojimenezp yes you can move on with the PR, appreciated!

**Comment by vvo** [comment]
@jrobber @pedrojimenezp closing for now, but feel free to reopen a PR / answer to comments I added

**Pull request closed without merging**